### PR TITLE
Feat/stacks cli

### PR DESCRIPTION
**Pull request by guillermogutierrez**
[XXXX-<Title> - Please use the Work Item number and Title as PR Name, not subtasks]

#### 📲 What

Support for the stacks cli v2.0 integration

🤔 Why
With the new version of the stacks cli we need to adapt the code to the new requirements.

🛠 How
Stacks cli expects a file called stackscli.yml file with all build steps. Also existing files require minor updates to ensure the scaffolding works as expected

#### 👀 Evidence

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [x] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: api-tests/pom.xml [hunk 1]
```diff
@@ -291,6 +292,84 @@
...
             </build>
         </profile>
+        <profile>
+            <id>archetype-resources</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>com.coderplus.maven.plugins</groupId>
+                        <artifactId>copy-rename-maven-plugin</artifactId>
+                        <version>1.0.1</version>
```
Comment (johnsall) [inline api-tests/pom.xml:302]: make a variable for the version?

File: java/pom.xml [hunk 2]
```diff
@@ -233,36 +235,36 @@
...
-<!--                <executions>-->
-<!--                    <execution>-->
-<!--                        <phase>integration-test</phase>-->
-<!--                        <goals>-->
-<!--                            <goal>verify</goal>-->
-<!--                        </goals>-->
-<!--                    </execution>-->
-<!--                </executions>-->
-<!--            </plugin>-->
+            <!--            <plugin>-->
```
Comment (johnsall) [inline java/pom.xml:238]: revert this?

File: stackscli.yml [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+      desc: Coping build and deploy pipelines into the project structure
+
+    # Build archetype for the main code
+    - action: cmd
+      cmd: mvn
+      args: clean archetype:create-from-project
+              -f ./java/pom.xml
+              -DpropertyFile=./java/archetype.properties
+              --quiet
+      desc: Creating the archetype resources using the web-api source code
```
Comment (johnsall) [inline stackscli.yml:53]: project reference, this is cqrs not web-api :)

File: stackscli.yml [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+              -DoutputDirectory={{ .Project.Directory.WorkingDir }}
+              -DarchetypeGroupId=com.amido.stacks.workloads
+              -DarchetypeArtifactId=stacks-api-cqrs-archetype
+              -DarchetypeVersion=1.0.0
+              -DgroupId=com.{{ .Input.Business.Company}}.{{ .Input.Business.Domain }}
+              -DartifactId={{ .Project.Name }}
+              -Dpackage=com.{{ .Input.Business.Company}}.{{ .Input.Business.Domain }}.{{.Input.Business.Component}}
+              --quiet
+              -B
+      desc: Generating the web-api project
```
Comment (johnsall) [inline stackscli.yml:75]: project ref again here

File: stackscli.yml (hunk 3 above)
Comment (guillermogutierrez) [reply stackscli.yml:53]: Nice spot

File: api-tests/pom.xml (hunk 1 above)
Comment (guillermogutierrez) [reply api-tests/pom.xml:302]: Done

File: java/pom.xml (hunk 2 above)
Comment (guillermogutierrez) [reply java/pom.xml:238]: I'm not sure what to do with the PACT configuration. We need to agree on this

**Pull request merged**